sphinx/builders/html.py

```diff
@@ -714,8 +728,9 @@ def index_page(self, pagename, doctree, title):
     def _get_local_toctree(self, docname, collapse=True, **kwds):
         if 'includehidden' not in kwds:
             kwds['includehidden'] = False
-        return self.render_partial(self.env.get_toctree_for(
-            docname, self, collapse, **kwds))['fragment']
+        toc = self.env.get_toctree_for(docname, self, collapse, **kwds)
+        self.fix_refuris(toc)
```

Comment:
This change will works only `docname==self.config.master_doc`.
Because self.fix_refuris() rewrites refuri attribute into `index.html#blabla` whether `docname` is 'index` or not.